selfdrive/debug/print_docs_diff.py

```diff
@@ -1,120 +1,93 @@
 #!/usr/bin/env python3
 import argparse
-from collections import defaultdict
-import difflib
-import pickle
-
-from openpilot.selfdrive.car.docs import get_all_car_docs
-from openpilot.selfdrive.car.docs_definitions import Column
-
-FOOTNOTE_TAG = "<sup>{}</sup>"
-STAR_ICON = '<a href="##"><img valign="top" ' + \
-            'src="https://media.githubusercontent.com/media/commaai/openpilot/master/docs/assets/icon-star-{}.svg" width="22" /></a>'
-VIDEO_ICON = '<a href="{}" target="_blank">' + \
-             '<img height="18px" src="https://media.githubusercontent.com/media/commaai/openpilot/master/docs/assets/icon-youtube.svg"></img></a>'
-COLUMNS = "|" + "|".join([column.value for column in Column]) + "|"
-COLUMN_HEADER = "|---|---|---|{}|".format("|".join([":---:"] * (len(Column) - 3)))
-ARROW_SYMBOL = "➡️"
-
-
-def load_base_car_docs(path):
-  with open(path, "rb") as f:
-    return pickle.load(f)
+import requests
+import re
 
+ARROW_SYMBOL = "➡️"
 
-def match_cars(base_cars, new_cars):
-  changes = []
-  additions = []
-  for new in new_cars:
-    # Addition if no close matches or close match already used
-    # Change if close match and not already used
-    matches = difflib.get_close_matches(new.name, [b.name for b in base_cars], cutoff=0.)
-    if not len(matches) or matches[0] in [c[1].name for c in changes]:
-      additions.append(new)
-    else:
-      changes.append((new, next(car for car in base_cars if car.name == matches[0])))
-
-  # Removal if base car not in changes
-  removals = [b for b in base_cars if b.name not in [c[1].name for c in changes]]
-  return changes, additions, removals
 
+def get_cars_docs_in_markdown(docs_content):
+  def _parse_markdown_table(md_table):
+    lines = md_table.strip().split('\n')
+    headers = [h.strip() for h in lines[0].split('|') if h]
+    assert len(headers) == len(set(headers)), 'Duplicate headers found in the markdown table'
+    return headers, [dict(zip(headers, [cell.strip() for cell in line.split('|')[1:-1]], strict=True)) for line in lines[2:]]
 
-def build_column_diff(base_car, new_car):
-  row_builder = []
-  for column in Column:
-    base_column = base_car.get_column(column, STAR_ICON, VIDEO_ICON, FOOTNOTE_TAG)
-    new_column = new_car.get_column(column, STAR_ICON, VIDEO_ICON, FOOTNOTE_TAG)
+  match = re.search(r"(\d+)\s+supported\s+cars\s*\n([\s\S]*?\|.*?\|[\s\S]*?\|[-:]+\|[\s\S]*?\n(?:[^\S\r\n]*\S.*\n)*\n?)", docs_content, re.IGNORECASE)
+  if not match:
+    raise RuntimeError("Couldn't find the car docs Markdown table.")
+  num_cars = int(match.group(1))
+  md_table = match.group(2).strip()
```

Comment:
```suggestion
  num_cars, md_table = match.groups()
```

I like unpacking in one line for the best readability, and you know immediately where the usage of `match` ends.